core/src/test/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotServiceSpec.scala

```diff
@@ -239,10 +242,198 @@ class SnapshotServiceSpec extends AnyWordSpecLike with Matchers with MockitoSuga
+                                                                ArgumentMatchers.eq(expectedPolicyInputs),
+                                                                any[RawlsRequestContext]
+        )
+    }
+
+    "not set a group-constraint policy on an existing WSM workspace if there is already a group-constraint policy even if the workspace has an auth domain" in withMinimalTestDatabase {
+      _ =>
+        val authDomainGroup = "authDomainGroup"
+        val mockSamDAO = defaultMockSamDao()
+        when(mockSamDAO.getResourceAuthDomain(any[SamResourceTypeName], any[String], any[RawlsRequestContext]))
+          .thenReturn(Future.successful(Seq(authDomainGroup)))
+
+        val existingPolicyInputs = new WsmPolicyInputs().inputs(
+          List(
+            new WsmPolicyInput()
+              .namespace("terra")
+              .name("group-constraint")
+              .additionalData(List(new WsmPolicyPair().key("group").value(authDomainGroup)).asJava)
+              .additionalData(List(new WsmPolicyPair().key("group").value("additionalPolicyGroup")).asJava)
+          ).asJava
+        )
+
+        val mockWorkspaceManagerDAO = defaultMockWorkspaceManagerDao()
+        when(mockWorkspaceManagerDAO.getWorkspace(any[UUID], any[RawlsRequestContext])).thenReturn(
+          new WorkspaceDescription().stage(WorkspaceStageModel.RAWLS_WORKSPACE).policies(existingPolicyInputs.getInputs)
+        )
+
+        val mockDataRepoDAO: DataRepoDAO = new MockDataRepoDAO("mockDataRepo")
+
+        val workspace = minimalTestData.workspace
+
+        val snapshotService = SnapshotService.constructor(
+          slickDataSource,
+          mockSamDAO,
+          mockWorkspaceManagerDAO,
+          "fake-terra-data-repo-dev",
+          mockDataRepoDAO
+        )(testContext)
+
+        // call createSnapshot on the service
+        Await.result(
+          snapshotService.createSnapshot(workspace.toWorkspaceName,
+                                         NamedDataRepoSnapshot(DataReferenceName("foo"),
+                                                               Option(DataReferenceDescriptionField("foo")),
+                                                               UUID.randomUUID()
+                                         )
+          ),
+          Duration.Inf
+        )
+
+        // assert that the service checked to see if the workspace exists
+        verify(mockWorkspaceManagerDAO).getWorkspace(
+          ArgumentMatchers.eq(workspace.workspaceIdAsUUID),
+          any[RawlsRequestContext]
+        )
+
+        // assert that the service DID NOT call WSM's createWorkspace
+        verify(mockWorkspaceManagerDAO, times(0)).createWorkspace(
+          any[UUID],
+          ArgumentMatchers.eq(WorkspaceType.RawlsWorkspace),
+          any[Option[WsmPolicyInputs]],
+          any[RawlsRequestContext]
+        )
+
+        // assert that the workspace's existing policies were not updated
```

Comment:
I'm trying to understand why we don't want them to be updated.. shouldn't the new policy just merge in?

If the answer is no because it would be an error case, then it seems like we should error if we encounter this, not just silently ignore the new policy.

Reply:
We don't want it to be updated because it's not Rawls's place to modify existing policies on a workspace in WSM (rawls-stage or MC).